### PR TITLE
Keep the script alive until all async operations are finished.

### PR DESCRIPTION
**Pull request by ericpromislow**
Not tied to a specific issue.

This is partly done in one of the commits in the v5 preferences work, but might as well get this done in a separate PR.

Signed-off-by: Eric Promislow <epromislow@suse.com>

### CODE REVIEW

**Comment by ericpromislow** [comment]
Doesn't work on linux:
```
Watching src/go/rdctl/pkg/options/generated/options.go
TypeError [ERR_FEATURE_UNAVAILABLE_ON_PLATFORM]: The feature watch recursively is unavailable on the current platform, which is being used to run Node.js
```

**Review by mook-as — CHANGES_REQUESTED** [review]
Do I understand things correctly that we don't _really_ care about watching the file, just that we need _something_ to hang around until everything is done?

In that case, it might make more sense to:
- `os.watch()` the root source directory (_without_ `recursive`, so that it works on Linux)
- Just do that as one giant `try` / `finally`.  Or, equivalently, `…().then(…).catch(…).then(<unwatch>)`.

It looks like quite a lot of the current code is for debugging?

**Comment by ericpromislow** [comment]
Yes, there was a lot of code for debugging in there, because we were getting invisible failures in the CI.

It turns out that an active `setTimeout` is enough to keep the script alive, so that's what I replaced the file watcher with.

File: scripts/postinstall.ts [hunk 1]
```diff
@@ -109,14 +110,25 @@ function buildDownloadContextFor(rawPlatform: DependencyPlatform, depVersions: D
...
-})
-  .catch((e) => {
-    console.error(e);
-    process.exit(1);
-  });
+// The main purpose of this setTimeout is to keep the script waiting until the main async function finishes
+const keepScriptAlive = setTimeout(() => { }, 24 * 3600 * 1000);
+
+(async() => {
+  const execFileP = util.promisify(execFile);
```
Comment (mook-as) [inline scripts/postinstall.ts:117]: It might be easier to just import `spawnFile` from `@pkg/utils/childProcess`?

File: scripts/postinstall.ts [hunk 2]
```diff
@@ -109,14 +110,25 @@ function buildDownloadContextFor(rawPlatform: DependencyPlatform, depVersions: D
...
+  let status = 2;
```
Comment (mook-as) [inline scripts/postinstall.ts:118]: Please have some comments on what this is, or rename it `exitCode` or something to make it more obvious. (There are many things that can be a status.)

**Review by mook-as — CHANGES_REQUESTED** [review]

File: scripts/postinstall.ts (hunk 1 above)
Comment (ericpromislow) [reply scripts/postinstall.ts]: Not exactly, but it's done.

File: scripts/postinstall.ts (hunk 2 above)
Comment (ericpromislow) [reply scripts/postinstall.ts]: renamed

**Review by mook-as — APPROVED** [review]
LGTM

**Review by mook-as — DISMISSED** [review]

**Comment by ericpromislow** [comment]
Just squashed the commits this review round.

**Review by mook-as — APPROVED** [review]

**Pull request merged**